From 900b75cd1e09768fd64ecc065171ea3885bb18fb Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 27 Jan 2014 21:58:42 +0000
Subject: [PATCH 019/102] run: Actually use timeout --foreground option
 (RHBZ#1025269).

The following commit managed to not actually add the --foreground
option to the timeout command, just test for it.  Add it this time.

  commit 681488877456b83f039dc518861f29ab4e1857f0
  Author: Richard W.M. Jones <rjones@redhat.com>
  Date:   Thu Dec 19 08:21:53 2013 +0000

    run: Use timeout --foreground option.

    If timeout doesn't have this option (RHEL 6) don't use timeout at all.

    Attempt to fix RHBZ#1025269.
(cherry picked from commit e3f72805d108c601280b889692c47d7d825e6079)
---
 run.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/run.in b/run.in
index c6bc9c4..a642d14 100755
--- a/run.in
+++ b/run.in
@@ -190,7 +190,7 @@ if timeout --help >/dev/null 2>&1; then
     if timeout --foreground 2 sleep 0 >/dev/null 2>&1; then
         # Does this version of timeout have the -k option?  (Not on RHEL 6)
         if timeout -k 10s 10s true >/dev/null 2>&1; then
-            timeout="timeout -k $timeout_kill $timeout_period"
+            timeout="timeout --foreground -k $timeout_kill $timeout_period"
         fi
     fi
 fi
-- 
1.9.3

